akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -73,8 +76,21 @@ trait AskSupport {
    *
    */
   def ask(actorRef: ActorRef, message: Any)(implicit timeout: Timeout): Future[Any] =
-    actorRef.internalAsk(message, timeout, ActorRef.noSender)
+    actorRef.internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
   def ask(actorRef: ActorRef, message: Any, sender: ActorRef)(implicit timeout: Timeout): Future[Any] =
+    actorRef.internalAsk(_ ⇒ message, timeout, sender)
+
+  /**
+   * A variant of `ask()` which allows to explicitly provide a reference to the actor
+   * which the result needs to be sent to in a message:
+   *
+   * {{{
+   *   ask(worker, respondTo => Message(respondTo))(timeout)
+   * }}}
+   */
+  def ask(actorRef: ActorRef, message: ActorRef ⇒ Any)(implicit timeout: Timeout): Future[Any] =
```

Comment:
I think it would be better to provide `ask(ActorRef, japi.Function)` now and address this problem after 2.12 release, so that `japi.Function` replacement with lambdas can be solved in a more uniform way across all of Java APIs. A single case of API which provides lambdas interop would introduce inconsistency, which is worse than potential ambiguity.